tests/providers/microsoft/azure/transfers/test_s3_to_wasb.py

```diff
@@ -0,0 +1,333 @@
+        # default parameters that are configured
+        operator = S3ToAzureBlobStorageOperator(
+            task_id=TASK_ID,
+            aws_conn_id="test-conn-id",
+            s3_bucket=S3_BUCKET,
+            s3_prefix=PREFIX,
+            container_name=CONTAINER_NAME,
+            blob_prefix=PREFIX,
+            replace=True,
+        )
+
+        # ... is None is used to validate if a value is None, while not ... is used to evaluate if a value
+        # is False
+        assert operator.task_id == TASK_ID
+        assert operator.aws_conn_id == AWS_CONN_ID
+        assert operator.wasb_conn_id == "wasb_default"
+        assert operator.s3_bucket == S3_BUCKET
+        assert operator.container_name == CONTAINER_NAME
+        assert operator.s3_prefix == PREFIX
+        assert operator.s3_key is None
+        assert operator.blob_prefix == PREFIX
+        assert operator.blob_name is None
+        assert not operator.create_container  # Should be false (match default value in constructor)
+        assert operator.replace
+        assert not operator.s3_verify  # Should be false (match default value in constructor)
+        assert operator.s3_extra_args == {}
+        assert operator.wasb_extra_args == {}
+
+    @mock.patch("airflow.providers.microsoft.azure.transfers.s3_to_wasb.S3Hook")
+    @mock.patch("airflow.providers.microsoft.azure.transfers.s3_to_wasb.WasbHook")
+    @mock.patch("tempfile.NamedTemporaryFile")
+    def test__execute__prefix_without_replace_empty_destination(
+        self, tempfile_mock, wasb_mock_hook, s3_mock_hook
+    ):
+        # Set the list files that the S3Hook should return, along with an empty list of files in the Azure
+        # Blob storage container. This scenario was picked for testing, as it's most likely the most common
+        # setting the operator will be used in
+        s3_mock_hook.return_value.list_keys.return_value = MOCK_FILES
+        wasb_mock_hook.return_value.get_blobs_list_recursive.return_value = []
+
+        s3_mock_hook.return_value.download_file.return_value = RawIOBase(b"test file contents")
+        tempfile_mock.return_value.__enter__.return_value.name = TEMPFILE_NAME
+
+        operator = S3ToAzureBlobStorageOperator(
+            task_id=TASK_ID,
+            s3_bucket=S3_BUCKET,
+            s3_prefix=PREFIX,
+            container_name=CONTAINER_NAME,
+            blob_prefix=PREFIX,
+        )
+        # Placing an empty "context" object here (using None)
+        uploaded_files = operator.execute(None)
+        assert sorted(uploaded_files) == sorted(MOCK_FILES)
+
+        # Using the default connection ID, along with the default value of verify (for the S3 hook)
+        s3_mock_hook.assert_called_once_with(aws_conn_id="aws_default", verify=False)
+        wasb_mock_hook.assert_called_once_with(wasb_conn_id="wasb_default")
+
+    # There are a number of very similar tests that use the same mocking, and for the most part, the same
```

Comment:
Thank you for your the feedback - I've gone back and updated the code to use parametrize.

Reply:
Well. the idea was to squash all the tests below to have one test and parameterized configuration for all the tests - effectively running all the tests below using single `test_` and array of parameters - this way we would not have to repeat the tests.

Reply:
Originally, there were nine different tests for the `get_files_to_move()` method. They covered three permutations of three possible uses. When I used parametrize to refactor the tests, reduced the number of tests to three, and parametrized each of those three tests (they're different enough that I wasn't comfortable squashing them all into a single test).

Reply:
Ah right. My bad. Can you please inline the variables instead of having constant ? I was mislead by that  - when you use parametrise it's best to just have the array of parameters inside the decorator, then you don't have to move up to the constant declaration to see what are the paremeters for testing.

Also Doc build needs to be fixed.

Reply:
Thank you for the feedback. I've updated the unit tests, as well as the docs.